cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/notifier/AlertaSelfHealingNotifierTest.java

```diff
@@ -0,0 +1,133 @@
+        failedBrokers.put(2, new SimpleDateFormat("yyyy-M-dd hh:mm:ssX").parse("2021-03-24 09:05:03Z").getTime());
+        Map<String, Object> parameterConfigOverrides = new HashMap<>(4);
+        parameterConfigOverrides.put(KAFKA_CRUISE_CONTROL_OBJECT_CONFIG, mockKafkaCruiseControl);
+        parameterConfigOverrides.put(ANOMALY_DETECTION_TIME_MS_OBJECT_CONFIG, 200L);
+        parameterConfigOverrides.put(FAILED_BROKERS_OBJECT_CONFIG, failedBrokers);
+        parameterConfigOverrides.put(BROKER_FAILURES_FIXABLE_CONFIG, true);
+        FAILURES = kafkaCruiseControlConfig.getConfiguredInstance(AnomalyDetectorConfig.BROKER_FAILURES_CLASS_CONFIG,
+                                                                  BrokerFailures.class,
+                                                                  parameterConfigOverrides);
+    }
+
+    @Test
+    public void testSlackAlertWithNoWebhook() {
+        _notifier = new MockAlertaSelfHealingNotifier(MOCK_TIME);
+        _notifier.alert(FAILURES, false, 1L, KafkaAnomalyType.BROKER_FAILURE);
+        assertEquals(0, _notifier.getAlertaMessageList().size());
+    }
+
+    @Test
+    public void testSlackAlertWithNoApiKey() {
+        _notifier = new MockAlertaSelfHealingNotifier(MOCK_TIME);
+        _notifier._alertaApiUrl = "http://dummy.alerta.api";
+        _notifier.alert(FAILURES, false, 1L, KafkaAnomalyType.BROKER_FAILURE);
+        assertEquals(0, _notifier.getAlertaMessageList().size());
+    }
+
+    @Test
+    public void testSlackAlertWithDefaultOptions() {
+        _notifier = new MockAlertaSelfHealingNotifier(MOCK_TIME);
+        _notifier._alertaApiUrl = "http://dummy.alerta.api";
+        _notifier._alertaApiKey = "dummy-alerta-api-key";
+        _notifier.alert(FAILURES, false, 1L, KafkaAnomalyType.BROKER_FAILURE);
+        assertEquals(2, _notifier.getAlertaMessageList().size());
+    }
+
+    @Test
+    public void testSlackAlertWithEnvironment() throws ParseException {
+        _notifier = new MockAlertaSelfHealingNotifier(MOCK_TIME);
+        _notifier._alertaApiUrl = "http://dummy.alerta.api";
+        _notifier._alertaApiKey = "dummy-alerta-api-key";
+        _notifier._alertaEnvironment = "MyEnv";
+        long time = new SimpleDateFormat("yyyy-M-dd hh:mm:ssX").parse("2021-03-24 10:02:01Z").getTime();
+        _notifier.alert(FAILURES, false, time, KafkaAnomalyType.BROKER_FAILURE);
+        assertEquals(2, _notifier.getAlertaMessageList().size());
+        assertEquals("BROKER_FAILURE", _notifier.getAlertaMessageList().get(0).getEvent());
+        assertEquals("BROKER_FAILURE", _notifier.getAlertaMessageList().get(1).getEvent());
+        assertEquals(_notifier._alertaEnvironment, _notifier.getAlertaMessageList().get(0).getEnvironment());
+        assertEquals("2021-03-24T10:01:01.000Z", _notifier.getAlertaMessageList().get(0).getCreateTime());
+        assertEquals("2021-03-24T09:05:03.000Z", _notifier.getAlertaMessageList().get(1).getCreateTime());
+    }
```

Comment:
Nit [Hardcoding]: Can we avoid hardcoding and instead use static final variable for test constants?